src/Single.php

```diff
@@ -60,6 +60,39 @@ public static function compress(iterable $data, iterable $selectors): \Generator
         }
     }
 
+    /**
+     * Return elements from the iterable only by given keys.
+     *
+     * Array of keys must contain unique items.
+     *
+     * @param iterable<mixed> $data
+     * @param array<mixed> $keys
+     * @param bool $strict
+     *
+     * @return \Generator
+     */
+    public static function compressAssociative(iterable $data, array $keys, bool $strict = true): \Generator
+    {
+        $map = iterator_to_array(self::reindex($keys, fn ($key) => UniqueExtractor::getString($key, $strict)));
+
+        yield from static::filterKeys($data, fn ($key) => isset($map[UniqueExtractor::getString($key, $strict)]));
```

Comment:
I think this is overly complicated, and it is solving for a use case that no one would expect. For the functionality with sets, it makes sense to have a strict mode because reasonable people would see `1`, `'1'`, and `true` as different _values_, and probably expect the default behavior to treat them as different things. 

However, when it comes to array and iterator keys in PHP, type coercion happens and that's just how the language works.

For example:

```php
php > @print_r([1 => 'first', '1' => 'second', true => 'third', 1.5 => 'fourth']);
Array
(
    [1] => fourth
)
```

And with an iterator:
```php
$it = new \ArrayIterator();
$it->offsetSet(1, 'first');
$it->offsetSet('1', 'second');
$it->offsetSet(true, 'third');

foreach ($it as $key => $value) {
    print("$key: $value");
}
```
`1: third`

Therefore, even with the tricky code to come up with unique keys, it can't work, because PHP does not retain the type of what the key actually was unless it was a string to begin with.

For example, based on how I think you are intending the functionality to work, if you add these test cases to `dataProviderForArrayStrict` in `CompressAssociativeTest`, I think you intended for this to work, but I think you will see that both test cases will fail:
```php
            [
                [1 => 'v1', '2' => 'v2'],
                [1, 2],
                [1],
                ['v1'],
            ],
            [
                [1 => 'v1', '2' => 'v2'],
                ['1', '2'],
                [2],
                ['v2'],
            ],
```

I do not think a "strict" mode makes sense for this function based on how PHP works.

Also, trying to support a notion of "composite keys" also isn't something PHP natively supports in arrays:
```php
php > $a = [[1] => 'v1'];
PHP Fatal error:  Illegal offset type in php shell code on line 1
```

In trying to make this function do everything, it might confuse users, and it is hard to reason about what the "correct" behavior should be. I recommend keeping this simple and have the same behavior regardless of whether the iterable is an array or iterator, and solve for the 99% use case.

If you remove the notion of "strict" for this function, then you could probably simplify the function to a simple foreach loop. Maybe something like this would work:
```php
$keyMap = \array_flip($keys);
foreach ($data as $key => $datum) {
    if (\array_key_exists($key, $keyMap)) {
        yield $key => $datum;
    }
}
```

PHP has limitations, and array/iterable keys is one of them. I don't recommend subverting expectations as it will probably only confuse users.

Reply:
Done.